### PR TITLE
CPU plugin does not work with dynamic input passed as ROI

### PR DESCRIPTION
**Pull request by vladislav-volkov**
### Details:
 - Fix for dynamic input passed as ROI in CPU plugin

### Tickets:
 - CVS-69815

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@mandrono Please review the PR

File: src/plugins/intel_cpu/src/mkldnn_infer_request.cpp [hunk 1]
```diff
@@ -432,12 +432,8 @@ void MKLDNNPlugin::MKLDNNInferRequest::SetBlob(const std::string& name, const In
             }
 
             if (blobDesc.getLayout() != InferenceEngine::Layout::ANY && foundInput->getTensorDesc().getLayout() != InferenceEngine::Layout::ANY) {
-                if (isDynamic && InferenceEngine::TensorDesc(foundInput->getPrecision(), blobDesc.getDims(), foundInput->getLayout()).getBlockingDesc() !=
-                        blobDesc.getBlockingDesc())
+                if (foundInput->getTensorDesc().getBlockingDesc().getOrder() != blobDesc.getBlockingDesc().getOrder())
                     IE_THROW(ParameterMismatch) << "Failed to set input blob. Layouts mismatch.";
-
-                if (!isDynamic && foundInput->getTensorDesc().getBlockingDesc() != blobDesc.getBlockingDesc())
```
Comment (mandrono) [inline src/plugins/intel_cpu/src/mkldnn_infer_request.cpp:439]: do we also have different strides in static case?

**Comment by dmitry-gorokhov** [comment]
@vladislav-volkov  @mandrono  Is the PR still actual. I guess we decided to include it into https://github.com/openvinotoolkit/openvino/pull/9456

**Comment by mandrono** [comment]
> @vladislav-volkov @mandrono Is the PR still actual? I guess we decided to include it into #9456

All necessary changes have already included https://github.com/openvinotoolkit/openvino/pull/9456. So this PR can be closed.

**Pull request closed without merging**